src/main/kotlin/SciviewBridge.kt

```diff
@@ -880,31 +899,36 @@ class SciviewBridge: TimepointObserver {
 
     /** Starts the sciview VR environment and optionally the eye tracking environment,
      * depending on the user's selection in the UI. Sends spot and track manipulation callbacks to the VR environment. */
-    fun launchVR(withEyetracking: Boolean = true) {
-        isVRactive = true
+    fun launchVR(withEyetracking: Boolean = true): Boolean {
+        // Test whether a headset is connected before starting sciview's VR launch routines
+        val hmd = OpenVRHMD(false, true)
+        if (!hmd.initializedAndWorking()) {
+            logger.warn("Could not find VR headset. Aborting launch of VR environment.")
+            hmd.close()
+            return false
+        }
+        hmd.close()
 
+        isVRactive = true
 
         thread {
-            if (withEyetracking) {
-                VRTracking = EyeTracking(sciviewWin)
-                (VRTracking as EyeTracking).run()
+            VRTracking = if (withEyetracking) {
```

Comment:
I'm a little unhappy with `VRTracking` being capitalised while being a variable. Could you change this to `vrTracking`? I'd also make sure that if it is null, you get an error message.